iothub_client/tests/common_dt_e2e/iothubclient_common_dt_e2e.c

```diff
@@ -651,6 +651,40 @@ void dt_e2e_get_complete_desired_test(IOTHUB_CLIENT_TRANSPORT_PROVIDER protocol,
 
     int initial_twin_version = dt_e2e_gettwin_version(serviceClientDeviceTwinHandle, deviceToUse);
 
+    // Receive first full twin before updating it.
+    time_t beginOperation, nowTime;
+    beginOperation = time(NULL);
+    while (
+        (nowTime = time(NULL)),
+        (difftime(nowTime, beginOperation) < MAX_CLOUD_TRAVEL_TIME) // time box
+        )
+    {
+        if (Lock(device->lock) != LOCK_OK)
+        {
+            ASSERT_FAIL("Lock failed");
+        }
+        else
+        {
+            if ((device->receivedCallBack) && (device->cb_payload != NULL))
```

Comment:
I'm not seeing where you reset `receivedCallBack`.  I think way code currently is, when you get to the "check that post-update twin has been received" it'll immediately see `receivedCallBack` as `true` instead of this being set in callback.

Reply:
ohhhh yes I do need to. Thank you!